expr/interval_compound_literal.go

```diff
@@ -0,0 +1,246 @@
+package expr
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/substrait-io/substrait-go/proto"
+	"github.com/substrait-io/substrait-go/types"
+)
+
+// IntervalCompoundLiteral creates an interval compound literal
+type IntervalCompoundLiteral struct {
+	years       int32
+	months      int32
+	days        int32
+	seconds     int32
+	subSeconds  int64
+	precision   types.TimePrecision
+	nullability types.Nullability
+}
+
+func (m IntervalCompoundLiteral) WithYears(years int32) IntervalCompoundLiteral {
+	return IntervalCompoundLiteral{
+		years:       years,
+		months:      m.months,
+		days:        m.days,
+		seconds:     m.seconds,
+		subSeconds:  m.subSeconds,
+		precision:   m.precision,
+		nullability: m.nullability,
+	}
+}
```

Comment:
All of these can be made simpler because you're using a value receiver.

```go
func (m IntervalCompoundLiteral) WithYears(years int32) IntervalCompoundLiteral {
   m.years = years
   return m
}
```

The value receiver is going to create a copy when this is called, so you don't need to explicitly make the copy yourself. Same can be done for the rest of the methods

Reply:
fixed it, Thanks